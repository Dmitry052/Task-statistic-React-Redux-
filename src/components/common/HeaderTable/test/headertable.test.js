import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { HeaderTable } from './../index';

configure({ adapter: new Adapter() });

describe('<HeaderTable />', () => {
  const itemTR = [
    {
      field: 'bill_id',
      displayName: 'BILL_ID',
      prefix: null,
      postfix: null,
      nowrap: false,
      path: 'bill_id',
      sortable: false,
      total: { text: 'Total:' },
      type: 'number'
    },

    {
      field: 'bill_date',
      displayName: 'BILL_DATE',
      prefix: null,
      postfix: null,
      nowrap: false,
      path: 'bill_date',
      sortable: false,
      total: { text: '-' },
      type: 'date'
    },

    {
      field: 'tokens_sold',
      displayName: 'TOKENS_SOLD',
      prefix: null,
      postfix: null,
      nowrap: false,
      path: 'tokens_sold',
      sortable: true,
      total: { text: '-' },
      type: 'number'
    }
  ];
  const name = 'bills';
  const index = 0;

  const tree = shallow(
    <HeaderTable itemTR={itemTR} name={name} index={index} />
  );

  it('should contain 2 tr elements', () => {
    expect(tree.find('tr').children().length).toBe(3);
  });

  it('Matches the snapshot HeaderTable component', () => {
    expect(toJson(tree)).toMatchSnapshot();
    tree.unmount();
  });
});
