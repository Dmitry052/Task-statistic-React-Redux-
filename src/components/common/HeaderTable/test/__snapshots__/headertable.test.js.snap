// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderTable /> Matches the snapshot HeaderTable component 1`] = `
<React.Fragment>
  <tr
    key="tr-bills-0"
  >
    <th
      colSpan={0}
      key="th-undefined"
      rowSpan={0}
    >
      BILL_ID
    </th>
    <th
      colSpan={0}
      key="th-undefined"
      rowSpan={0}
    >
      BILL_DATE
    </th>
    <th
      colSpan={0}
      key="th-undefined"
      rowSpan={0}
    >
      TOKENS_SOLD
      <WithStyles(Sort)
        field="tokens_sold"
        path="tokens_sold"
        type="number"
      />
    </th>
  </tr>
</React.Fragment>
`;
